src/main/java/ui/MenuControl.java

```diff
@@ -87,7 +87,45 @@ private Menu createFileMenu() {
         return file;
 
     }
-    
+
+    /**
+     * Creates an empty board. User will be prompted to
+     * confirm the action if there are unclosed panels.
+     */
+    private void onBoardNew() {
+        logger.info("Menu: Boards > New");
+
+        if (!isNewBoardCreationDialogConfirmed()) {
+            logger.info("New board creation cancelled");
+            return;
+        }
+
+        panels.closeAllPanels();
+        onBoardSaveAs();
+    }
+
+    /**
+     * Returns a boolean indicating whether we can proceed with
+     * the new board creation. If there are no panels open, it
+     * returns true; else a dialog is shown for the user to confirm.
+     */
+    private boolean isNewBoardCreationDialogConfirmed() {
+        List<PanelInfo> panelList = panels.getCurrentPanelInfos();
+        if (!panelList.isEmpty()) {
+            Alert dlg = new Alert(AlertType.CONFIRMATION, "");
```

Comment:
I think @damithc meant this instead:

```java
if (panelList.isEmpty()) {
    return true;
}
// everything else
```
Makes the logic a bit simpler to read.